src/Instrumentation/Psr14/src/Psr14Instrumentation.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\Instrumentation\Psr14;
+
+use Composer\InstalledVersions;
+use OpenTelemetry\API\Instrumentation\CachedInstrumentation;
+use OpenTelemetry\API\Trace\Span;
+use OpenTelemetry\API\Trace\StatusCode;
+use OpenTelemetry\Context\Context;
+use function OpenTelemetry\Instrumentation\hook;
+use OpenTelemetry\SemConv\TraceAttributes;
+use Psr\EventDispatcher\EventDispatcherInterface;
+
+use Throwable;
+
+/**
+ * @psalm-suppress ArgumentTypeCoercion
+ */
+class Psr14Instrumentation
+{
+    public const NAME = 'psr14';
+
+    public static function register(): void
+    {
+        $instrumentation = new CachedInstrumentation(
+            'io.opentelemetry.contrib.php.psr14',
+            InstalledVersions::getVersion('open-telemetry/opentelemetry-auto-psr14'),
```

Comment:
Does this depend on composer 2, per [these docs](https://getcomposer.org/doc/07-runtime.md#installed-versions)?

I am not sure if this would break with composer v1.x, so not adding as a blocker.

Reply:
I got this from the `\OpenTelemetry\Contrib\Instrumentation\ExtAmqp\ExtAmqpInstrumentation`, for here I'll add the required dependencies.

Reply:
pushed a change for this

Reply:
Partially resolving this with an additional PR https://github.com/open-telemetry/opentelemetry-php-contrib/pull/224